### PR TITLE
[#165718791] Track instabug report on mixpanel

### PR DESCRIPTION
**Pull request by Undermaken**
this PR aims to track on mixpanel when an instabug report is opened and closed.
When a report is opened an event is sent to mixpanel by attaching report **type** (i.e chat or bug)
When a report is closed an event is sent to mixpanel by attaching report **type** and **how** the report is closed (i.e cancel, submit etc.)

This PR has no visible effects to the end user

### CODE REVIEW

File: ts/components/InstabugButtons.tsx [hunk 1]
```diff
@@ -3,34 +3,53 @@ import * as React from "react";
...
+      }
+    );
+  }
+
   public render() {
     return (
       this.props.isDebugModeEnabled && (
         <React.Fragment>
-          <Button onPress={this.handleIBChatPress} transparent={true}>
+          <Button onPress={() => this.handleIBChatPress()} transparent={true}>
```
Comment (fpersico) [inline ts/components/InstabugButtons.tsx]:
```suggestion
          <Button onPress={this.handleIBChatPress} transparent={true}>
```

File: ts/components/InstabugButtons.tsx [hunk 2]
```diff
@@ -3,34 +3,53 @@ import * as React from "react";
...
             <IconFont name="io-chat" color={this.props.color} />
           </Button>
-          <Button onPress={this.handleIBBugPress} transparent={true}>
+          <Button onPress={() => this.handleIBBugPress()} transparent={true}>
```
Comment (fpersico) [inline ts/components/InstabugButtons.tsx]:
```suggestion
          <Button onPress={this.handleIBBugPress} transparent={true}>
```

File: ts/store/actions/debug.ts [hunk 3]
```diff
@@ -1,10 +1,41 @@
...
  */
 export const setDebugModeEnabled = createStandardAction(
   "DEBUG_MODE_SET_ENABLED"
 )<boolean>();
 
-export type DebugActions = ActionType<typeof setDebugModeEnabled>;
+/**
+ * An Instabug report is open
+ */
+export const openInstabugReport = createStandardAction("INSTABUG_REPORT_OPEN")<
```
Comment (fpersico) [inline ts/store/actions/debug.ts]:
```suggestion
export const instabugReportOpened = createStandardAction("INSTABUG_REPORT_OPENED")<
```

File: ts/store/actions/debug.ts [hunk 4]
```diff
@@ -1,10 +1,41 @@
...
+  InstabugReport
+>();
+
+/**
+ * An Instabug report is closed
+ */
+export const closeInstabugReport = createStandardAction(
```
Comment (fpersico) [inline ts/store/actions/debug.ts]:
```suggestion
export const instabugReportClosed = createStandardAction(
```

File: ts/store/middlewares/analytics.ts [hunk 5]
```diff
@@ -188,6 +189,17 @@ const trackAction = (mp: NonNullable<typeof mixpanel>) => (
       return mp.track(action.type, action.payload);
     }
 
+    // instabug
+    case getType(openInstabugReport):
+      return mp.track(action.type, {
```
Comment (fpersico) [inline ts/store/middlewares/analytics.ts]:
```suggestion
      return mp.track(action.type, payload)
```

File: ts/store/middlewares/analytics.ts [hunk 6]
```diff
@@ -188,6 +189,17 @@ const trackAction = (mp: NonNullable<typeof mixpanel>) => (
...
+        type: action.payload.type
+      });
+    case getType(closeInstabugReport):
+      return mp.track(action.type, {
```
Comment (fpersico) [inline ts/store/middlewares/analytics.ts]:
```suggestion
      return mp.track(action.type, payload)
```

File: ts/components/InstabugButtons.tsx (hunk 1 above)
Comment (Undermaken) [reply ts/components/InstabugButtons.tsx]: @fpersico may arrow function needed because in that function (_handleIBChatPress_) the code refers to **this**?

Comment (fpersico) [reply ts/components/InstabugButtons.tsx]: You have to define the function like

```
private handleIBChatPress = () => {
    this.props.dispatchIBReportOpen("chat");
    Chats.show();	    Chats.show();
  }	  }
``

**Comment by fpersico** [comment]
@Undermaken please fix this PR.

**Comment by Undermaken** [comment]
> @Undermaken please fix this PR.
@fpersico 
I tried to fix the instabug patch generating it few times, but the patch it's always the same. Locally, on my machine, it works and It is created automatically by **patch-package**
Any suggestions to find the issue?

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#165718791](https://www.pivotaltracker.com/story/show/165718791): Come DEV voglio avere un evento mixpanel quando l'utente invia una segnalazione di bug o apre una chat

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**